source/darwin/secure_transport_tls_channel_handler.c

```diff
@@ -321,6 +325,10 @@ static void s_invoke_negotiation_callback(struct aws_channel_handler *handler, i
 static int s_drive_negotiation(struct aws_channel_handler *handler) {
     struct secure_transport_handler *secure_transport_handler = handler->impl;
 
+    if (secure_transport_handler->stats.handshake_status == AWS_MTLS_STATUS_NONE) {
```

Comment:
Should we assert if it's in another state? Is that even possible?

Reply:
I guess we could verify it's always in either None or Ongoing.